src/main.h

```diff
@@ -109,16 +106,13 @@ static const unsigned int MAX_REJECT_MESSAGE_LENGTH = 111;
 /* Maximum number of heigths meaningful when looking for block finality */
 static const int MAX_BLOCK_AGE_FOR_FINALITY = 2000;
 
-#ifdef ENABLE_ADDRESS_INDEXING
+static const bool DEFAULT_TXINDEX = false;
+static const bool DEFAULT_MATURITYHEIGHTINDEX = false;
 static const bool DEFAULT_ADDRESSINDEX = false;
 static const bool DEFAULT_TIMESTAMPINDEX = false;
 static const bool DEFAULT_SPENTINDEX = false;
 static const unsigned int DEFAULT_DB_MAX_OPEN_FILES = 1000;
 static const bool DEFAULT_DB_COMPRESSION = true;
-#else
```

Comment:
In my opinion the values for `DEFAULT_DB_MAX_OPEN_FILES` and `DEFAULT_DB_COMPRESSION` should stay respectively `64` and `false` as per previous non-ENABLE_ADDRESS_INDEXING scenario; maybe two additional constants can be provided (for example `DEFAULT_DB_MAX_OPEN_FILES_ADDRESS_INDEXING` and `DEFAULT_DB_COMPRESSION_ADDRESS_INDEXING`), their values being used if `"-addressindex"` arg is enabled.

Reply:
On the matter please see https://github.com/bitcoin/bitcoin/pull/12495
The 64 value for MAX_DB_OPEN_FILES we enforced has a serious impact on performances and gives negligible advantage in memory savings.

Agree on enabling compression only for explicit user choice.

Reply:
> Notice that bitcoin currently is still using 64 on non-windows 32bit systems.
Given that we also support linux 32bit, at least in theory, maybe it makes sense to do something similar?

I wonder nowadays who dares to rely on a 32 bit OS ... however the suggestion makes sense.

> On the compression, I'm wondering what happens if we disable compression on a local db that was created with compression? Maybe it makes sense to keep the defaults as they were before this patch, that is enable compression if a node is started with -addressindex?

I've tested that and the answer is : nothing happens. I'll explain:
Compression involves only  ldb files `.zen/blocks/index` data directory and the parameter is relevant **only on flushing file to disk**. The compression info is written in the header of the file so when ldb have to reopen it it firstly read the meta info of the file and decide whether the content is compressed.
So basically this happens : start an instance with indexing and compression enabled causes newly generated files to be written to disk as compressed. Then stop the instance and restart with indexing but this time with compression disabled : newly generated files are not compressed but existing one still are and when ldb has to open/modify those it reads the compression flag from file's metadata not from the setting in the environment.

Reply:
@AndreaLanfranchi please correct me if I'm wrong, but from what I can see I believe that this PR is changing the defaults in terms of compression.
Before this PR we have compression enabled by default for `addressindex`, and disabled otherwise. After this PR, we have compression disabled in all cases by default.

If the premise is correct, I think it would be better to restore the previous defaults in terms of compression, based on the runtime switch `-addressindex` (and probably also `-spentindex` and `-txindex`?).

Reply:
This is not completely correct
It was the preprocessor directive `ENABLE_ADDRESS_INDEXING` which was setting compression to true, not the CLI argument `addressindex` hence compression was enabled regardless the user choice. (i.e. even if no -addressindex was set in instance launch)

Reply:
Good point, I agree.

Still, my assumption is that the main use of the addressindex branch (i.e. the bitcore binary packages) is indeed with the `-addressindex` flag enabled at runtime. If that is correct, I am suggesting to minimize the number of users with different compression settings after this PR.

Also, I'm not sure about the rationale behind the choice of defaults for the compression, but I guess that the main reason is that addressindexing nodes store a greater amount of data, hence they want compression to be enabled to limit the db size on the storage. If this is correct, the impact of this PR for current bitcore nodes running without `-addressindex` (limited db size increase) would probably be smaller than the impact of disabling compression for actual addressindexing nodes (bigger db size increase).

With that being said, the ideal thing to do would probably be measuring the actual impact of compression on the db size and on access speed, and decide accordingly.

Reply:
I honestly don't see the point here.

Default db compression is set to false which adds no compression/decompression overhead (useful in most cases when users DO NOT use indexing features).
I have a fully synced node (at the time of this writing with this PR codebase) with all indexes enabled and no compression and the size of  `.zen/blocks/index` (the only one affected by compression) is 17 GiBi (total data size is roughly 82GiBi)

If that amount of space becomes an issue for some usecases the owner of the instance can freely enable the compression by adding `-blocktreedbcompression` argument to command line.

This IMHO adheres more to the concept of "optional feature"

Reply:
Prior of this PR instead the interpretation was ambiguous as the compression was ON by default for builds with the directive set and if user did not want the index files to be compressed had to add `-noblocktreedbcompression` which is to me a bit of nonsense.

Reply:
As additional information I issued a reindex of the sync'ed node (block 1.27M roughly) with compression enabled as :

```
./src/zend -txindex -spentindex -timespentindex -addressindex -maturityheightindex -reindex -blocktreedbcompression 
```

After 15 hours the reindexing is still at block 3.9M with less than 10M total transactions indexed (at the tip total transactions are slightly more than 90M) and the total usage `.zen/blocks/index` is 2.8 GiBi.
As a comparative metric consider (on same hardware)  the full sync from scratch (hence including blocks download from peers) with compression disabled and **all indexes enabled** took about 18 hours.